### PR TITLE
feature: install kubesphere on Civo Cloud

### LINKED ISSUE #291: Add KubeSphere to the Civo Marketplace
This issue is a:
* [ ] Bug Report
* [ ] Enhancement suggestion
* [x] Feature request

Hi Civo Team, 

[KubeSphere](http://kubesphere.io/) is an open-source container platform built on Kubernetes, it also supports K3s as the underlying orchestration system. We would like to add KubeSphere to the Civo Marketplace, enabling users to easily manage cloud-native applications and create CI/CD pipelines in multi-cluster and multi-cloud environments.

Thanks.

### PR DESCRIPTION
**Pull request by mangoGoForward**
Thank you for wanting to submit a Pull Request to the Civo Kubernetes Marketplace repository!

Resolved #291

If your pull request is to submit a new application to the marketplace, please answer the following questions:

* [x] Have you followed the guidelines in our Contributing document in setting up the application?
* [x] Have you checked that the application your submission is proposing to add does not yet exist on the Marketplace, or as a pull request to be processed?
* [x] Have you tested your application on the Civo managed Kubernetes service? Please include a screenshot.
* [x] Can you confirm that the software you are submitting licensed for use in a publicly-available Kubernetes marketplace?

The screenshot like below:

1. Add application to cluster

image

2. Get pod in specified namespace

image

3. Visit the KubeSphere console

image

Please have a look if you have free time. @saiyam1814 @FeynmanZhou

### CODE REVIEW

**Comment by saiyam1814** [comment]
@mangoGoForward I have left a comment - a small query I had

**Comment by mangoGoForward** [comment]
> @mangoGoForward I have left a comment - a small query I had

Hi @saiyam1814 , I haven't received any messages,  if you have free time, please point out to help me improve.

File: kubesphere/install.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+# Init version
+export RELEASE=v3.2.1
+
+# Install clusterconfiguration crd
+kubectl apply -f https://github.com/kubesphere/ks-installer/releases/download/$RELEASE/cluster-configuration.yaml
```
Comment (saiyam1814) [inline kubesphere/install.sh:7]: WHy this is separate and app.yaml is separate ?
CRD is also there in app.yaml , is that different from this one ?

Comment (saiyam1814) [reply kubesphere/install.sh:7]: @mangoGoForward

Comment (mangoGoForward) [reply kubesphere/install.sh:7]: Kubesphere supports enable the pluggable components, and the `ClusterConfiguration` in the `install.sh` is a cr to control which components should be install, it must be wait the CRD of `clusterconfigurations.installer.kubesphere.io` in the `app.yaml` applied in k8s successful.

Comment (saiyam1814) [reply kubesphere/install.sh:7]: ok Also 
app.yaml will be applied first and then the install.sh will run. Is that fine for above ?

Comment (mangoGoForward) [reply kubesphere/install.sh:7]: Yup, and we applied a remote file in `install.sh`, should we change to apply a stdin?

Comment (saiyam1814) [reply kubesphere/install.sh:7]: As long as you can maintain the remote file should be fine

Comment (FeynmanZhou) [reply kubesphere/install.sh:7]: The remote YAML file will be updated and maintained all the time. We might only need to update the version variable in this repo.

Comment (mangoGoForward) [reply kubesphere/install.sh:7]: yeah

**Comment by saiyam1814** [comment]
LGTM

**Comment by FeynmanZhou** [comment]
Hi @pixiake, 

As Saiyam finds that the port of the KubeSphere node (9100) exporter has a conflict with the built-in traefik, could you please help us to update the port number in ks-installer?

**Comment by pixiake** [comment]
> Hi @pixiake,
> 
> As Saiyam finds that the port of the KubeSphere node (9100) exporter has a conflict with the built-in traefik, could you please help us to update the port number in ks-installer?

I have submitted a pr to support customizing node-exporter's port. Once this pr is merged, I will update a tag and image for ks-installer.

https://github.com/kubesphere/ks-installer/pull/1909

**Comment by pixiake** [comment]
@mangoGoForward @FeynmanZhou 

The pull request that supports custom node-exporter's ports has been merged, and I have updated the tag and image of ks-installer.
https://github.com/kubesphere/ks-installer/tree/v3.2.1-ae451377/deploy

### How to use:
Add `spec.monitoring.node-exporter.port: xxx` to the configuration file.

```
spec:
  ...
  monitoring:
    node_exporter:
      port: 9100
  ...
```

**Comment by saiyam1814** [comment]
Great! Drop me a ping once the PR for the marketplace is in :)

**Pull request merged**